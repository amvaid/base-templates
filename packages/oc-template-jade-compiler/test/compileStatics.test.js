jasmine.DEFAULT_TIMEOUT_INTERVAL = 20000;

const path = require('path');
const fs = require('fs-extra');
const nodeDir = require('node-dir');

const compileStatics = require('../lib/compileStatics.js');
const componentPath = path.join(__dirname, '../../../mocks/jade-component');
const publishPath = path.join(componentPath, '_package');
const publishFileName = 'template.js';
const withStatic = (staticFiles, minify) => ({
  componentPackage: {
    oc: {
      files: {
        static: staticFiles
      }
    }
  },
  publishPath,
  componentPath,
  minify: minify || false
});

afterEach(() => {
  fs.emptyDirSync(publishPath + '/assets');
});

test('when oc.files.static is empty', done => {
  compileStatics(withStatic([]), (error, result) => {
    expect(error).toBeNull();
    expect(result).toBe('ok');
    done();
  });
});

test('when oc.files.static contains a folder that doesnt exist', done => {
  compileStatics(withStatic(['src']), (error, result) => {
    expect(error).toContain('not found');
    expect(result).toBeUndefined();
    done();
  });
});

test('when oc.files.static contain reference to a non-folder', done => {
  compileStatics(withStatic(['template.jade']), (error, result) => {
    expect(error).toContain('must be a directory');
    expect(result).toBeUndefined();
    done();
  });
});

test('compile statics when oc.files.static contains valid folder and minify is false', done => {
  compileStatics(withStatic(['assets']), (error, result) => {
    expect(error).toBeNull();
    expect(result).toBe('ok');
    nodeDir.paths(publishPath + '/assets', (err, res) => {
      const files = res.files.sort();
      expect(
        files.map(file => path.relative(__dirname, file))
      ).toMatchSnapshot();
      files.forEach(file => {
        if (!/.png$/.test(file)) {
          expect(fs.readFileSync(file, 'UTF8')).toMatchSnapshot();
        }
      });
      done();
    });
  });
});

test('compile statics when oc.files.static contains valid folder and minify is true', done => {
  const minify = true;
  compileStatics(withStatic(['assets'], minify), (error, result) => {
    expect(error).toBeNull();
    expect(result).toBe('ok');
    nodeDir.paths(publishPath + '/assets', (err, res) => {
      const files = res.files.sort();
      expect(
        files.map(file => path.relative(__dirname, file))
      ).toMatchSnapshot();
      files.forEach(file => {
        if (!/.png$/.test(file)) {
          expect(fs.readFileSync(file, 'UTF8')).toMatchSnapshot();
        }
      });
      done();
    });
  });
});
