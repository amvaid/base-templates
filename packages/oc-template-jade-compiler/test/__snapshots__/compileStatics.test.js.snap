// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`compile statics when oc.files.static contains valid folder and minify is false 1`] = `
Array [
  "../../../mocks/jade-component/_package/assets/css/style.css",
  "../../../mocks/jade-component/_package/assets/img/test.png",
  "../../../mocks/jade-component/_package/assets/js/lib.js",
]
`;

exports[`compile statics when oc.files.static contains valid folder and minify is false 2`] = `
"body {
  padding-left: 11em;
  font-family: Georgia, \\"Times New Roman\\", Times, serif;
  color: purple;
  background-color: #d8da3d
}
ul.navbar {
  list-style-type: none;
  padding: 0;
  margin: 0;
  position: absolute;
  top: 2em;
  left: 1em;
  width: 9em
}
h1 {
  font-family: Helvetica, Geneva, Arial, SunSans-Regular, sans-serif
}
ul.navbar li {
  background: white;
  margin: 0.5em 0;
  padding: 0.3em;
  border-right: 1em solid black
}
ul.navbar a {
  text-decoration: none
}
a:link {
  color: blue
}
a:visited {
  color: purple
}"
`;

exports[`compile statics when oc.files.static contains valid folder and minify is false 3`] = `
"const text = 'Hello World';
const say = something => console.log(something);
say(text);
"
`;

exports[`compile statics when oc.files.static contains valid folder and minify is true 1`] = `
Array [
  "../../../mocks/jade-component/_package/assets/css/style.css",
  "../../../mocks/jade-component/_package/assets/img/test.png",
  "../../../mocks/jade-component/_package/assets/js/lib.js",
]
`;

exports[`compile statics when oc.files.static contains valid folder and minify is true 2`] = `"body{padding-left:11em;font-family:Georgia,\\"Times New Roman\\",Times,serif;color:purple;background-color:#d8da3d}ul.navbar{list-style-type:none;padding:0;margin:0;position:absolute;top:2em;left:1em;width:9em}h1{font-family:Helvetica,Geneva,Arial,SunSans-Regular,sans-serif}ul.navbar li{background:#fff;margin:.5em 0;padding:.3em;border-right:1em solid #000}ul.navbar a{text-decoration:none}a:link{color:#00f}a:visited{color:purple}"`;

exports[`compile statics when oc.files.static contains valid folder and minify is true 3`] = `"var text=\\"Hello World\\",say=function(o){return console.log(o)};say(text);"`;
